### PR TITLE
For #8107 feat(nimbus): add advanced targeting options for mobile behavioral targeting

### PR DESCRIPTION
**Pull request by jeddai**
Because...

    The behavioral event store now exists within Firefox iOS and Android

This commit...

    Adds advanced targeting options that leverage the events being recorded.

**A note:** We need two distinct targeting configs for each behavior, one for iOS and one for Android. These are required because the same event is named differently across the two apps. Something I noticed is Experimenter seems to be using the targeting config `slug` field split+capitalized for displaying the config, and I don't think we'd want "Android" and "Ios" showing up in the list. If we do, something worth noting is that "Ios" should be "iOS".

iOS set as the app
image

Android set as the app
image

### CODE REVIEW

File: app/experimenter/targeting/constants.py [hunk 1]
```diff
@@ -868,6 +868,56 @@ def __post_init__(self):
     application_choice_names=[a.name for a in Application],
 )
 
+CORE_ACTIVE_USERS_TARGETING = "'{event}'|eventsCountNonZero('Days', 28, 0) >= 21"
```
Comment (yashika-khurana) [inline app/experimenter/targeting/constants.py:871]: nitpick: Can you move these constants to the top 🆙

File: app/experimenter/targeting/constants.py [hunk 2]
```diff
@@ -868,6 +868,56 @@ def __post_init__(self):
...
+    slug="ios_core_active_users",
+    description="Users who have been active at least 21 out of the last 28 days",
+    targeting=CORE_ACTIVE_USERS_TARGETING.format(event="app_cycle.foreground"),
+    desktop_telemetry="",
+    sticky_required=True,
+    is_first_run_required=False,
+    application_choice_names=(Application.IOS.name, Application.FOCUS_IOS.name),
+)
+
+RECENTLY_LOGGED_IN_USERS_TARGETING = "'{event}'|eventsCountNonZero('Weeks', 12, 0) >= 1"
```
Comment (yashika-khurana) [inline app/experimenter/targeting/constants.py:895]: same as before to move this 🆙

**Review by yashika-khurana — CHANGES_REQUESTED** [review]
Meanwhile , just a nit pick

File: app/tests/integration/nimbus/conftest.py [hunk 3]
```diff
@@ -141,7 +141,7 @@ def experiment_url(base_url, default_data, slugify):
 
 @pytest.fixture
 def experiment_name(request):
-    return f"{request.node.name[:75]}-{str(uuid.uuid4())[:4]}"
+    return f"{request.node.name[:75]}{str(uuid.uuid4())[:4]}"
```
Comment (jrbenny35) [inline app/tests/integration/nimbus/conftest.py:144]: 1 char too much

**Review by jrbenny35 — APPROVED** [review]

File: app/tests/integration/nimbus/conftest.py (hunk 3 above)
Comment (yashika-khurana) [reply app/tests/integration/nimbus/conftest.py:144]: 😆

**Review by yashika-khurana — APPROVED** [review]

**Pull request merged**